### PR TITLE
Replaced manual truncates in run() with truncateAll()

### PR DESCRIPTION
**Pull request by AlexVanderbist**
As described in spatie/blender/issues/299

### CODE REVIEW

File: src/DatabaseSeeder.php [hunk 1]
```diff
@@ -29,13 +29,26 @@ public function run()
...
-        $this->truncateContentBlocksTable();
-        $this->truncateActivityTable();
+        $this->truncateAll();
 
         $this->clearMediaDirectory();
     }
 
+    protected function truncateAll()
+    {
+        Schema::disableForeignKeyConstraints();
```
Comment (freekmurze) [inline src/DatabaseSeeder.php:39]: Can be replaced with

```php
$tableDropper = TableDropperFactory::create(DB::getDriverName());

$tableDropper->dropAllTables();

DB::reconnect();
```

Comment (sebastiandedeyne) [reply src/DatabaseSeeder.php:39]: It can't, the TableDropper drops tables, we need to truncate.

Comment (freekmurze) [reply src/DatabaseSeeder.php:39]: True, sorry for the noise

File: src/DatabaseSeeder.php [hunk 2]
```diff
@@ -29,13 +29,26 @@ public function run()
...
+
+        collect(DB::select("SHOW FULL TABLES WHERE Table_Type = 'BASE TABLE'"))
+            ->map(function ($tableProperties) {
+                return get_object_vars($tableProperties)[key($tableProperties)];
+            })
+            ->each(function (string $tableName) {
```
Comment (sebastiandedeyne) [inline src/DatabaseSeeder.php]: Could you add an exception for `migrations`?

File: src/DatabaseSeeder.php [hunk 3]
```diff
@@ -29,13 +29,26 @@ public function run()
...
+                DB::table($tableName)->truncate();
+            });
+
+        Schema::enableForeignKeyConstraints();
+    }
+
     protected function truncate(string ...$tables)
```
Comment (sebastiandedeyne) [inline src/DatabaseSeeder.php]: Keep this for backwards compat, but mark it as deprecated.

**Review by sebastiandedeyne — CHANGES_REQUESTED** [review]

**Review by sebastiandedeyne — APPROVED** [review]

**Comment by sebastiandedeyne** [comment]
Thanks!

**Pull request merged**